# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGL2ContextSync.cpp
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGL2ContextSync.cpp
--- a/dom/canvas/WebGL2ContextSync.cpp	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGL2ContextSync.cpp	Wed Jan 08 22:19:14 2020 +0000
@@ -107,22 +107,19 @@
   gl->fWaitSync(sync.mGLName, flags, LOCAL_GL_TIMEOUT_IGNORED);
 }
 
-void WebGL2Context::GetSyncParameter(JSContext*, const WebGLSync& sync,
-                                     GLenum pname,
-                                     JS::MutableHandleValue retval) {
+MaybeWebGLVariant WebGL2Context::GetSyncParameter(const WebGLSync& sync,
+                                                  GLenum pname) {
   const FuncScope funcScope(*this, "getSyncParameter");
-  retval.setNull();
-  if (IsContextLost()) return;
+  if (IsContextLost()) return Nothing();
 
-  if (!ValidateObject("sync", sync)) return;
+  if (!ValidateObject("sync", sync)) return Nothing();
 
   ////
 
   const bool canBeAvailable =
       (sync.mCanBeAvailable || StaticPrefs::webgl_allow_immediate_queries());
   if (!canBeAvailable && pname == LOCAL_GL_SYNC_STATUS) {
-    retval.set(JS::Int32Value(LOCAL_GL_UNSIGNALED));
-    return;
+    return AsSomeVariant(LOCAL_GL_UNSIGNALED);
   }
 
   GLint result = 0;
@@ -137,12 +134,11 @@
         sync.MarkSignaled();
       }
 
-      retval.set(JS::Int32Value(result));
-      return;
+      return AsSomeVariant(result);
 
     default:
       ErrorInvalidEnumInfo("pname", pname);
-      return;
+      return Nothing();
   }
 }
 